### DISCUSSION TITLE
Procreate Procreate Procreate

### DISCUSSION THREAD

**Discussion by ramzichamcham** [keiwando/toonsquid#322] [Ideas]
I use procreate and I love procreate. Now I also love Toonsquid. I want to be able to draw my assets in procreate and use them in Toonsquid more easily. The export process is a bit of hassle but it’s ok. My biggest request is to be able to conserve the layers from procreate in the drawing layers in toonsquid. An ideal workflow would be: export from procreate in .procreate or PSD format (if that conserves the layers) directly to toonsquid where the assets automatically get added to the toonsquid libray, and I can directly work on them in the Drawing layers pane. 
If for example I want to create a walk cycle in toonsquid, let’s say I create all the body parts on different layers in one project in procreate. In procreate I would have to hide all the layers and background and keep only the body part that I want, and export that into a folder in my files and then from toonsquid to add it to the library. I have to do this for all the parts of my animation when they are all already drawn and on separate layers/groups. This is my biggest problem right now with toonsquid, I’m trying to find a workaround, if there is one, please share it! Thanks!

**Comment by VeiloPot** [comment]
> I use procreate and I love procreate. Now I also love Toonsquid. I want to be able to draw my assets in procreate and use them in Toonsquid more easily. The export process is a bit of hassle but it’s ok. My biggest request is to be able to conserve the layers from procreate in the drawing layers in toonsquid. An ideal workflow would be: export from procreate in .procreate or PSD format (if that conserves the layers) directly to toonsquid where the assets automatically get added to the toonsquid libray, and I can directly work on them in the Drawing layers pane. 
> 
> If for example I want to create a walk cycle in toonsquid, let’s say I create all the body parts on different layers in one project in procreate. In procreate I would have to hide all the layers and background and keep only the body part that I want, and export that into a folder in my files and then from toonsquid to add it to the library. I have to do this for all the parts of my animation when they are all already drawn and on separate layers/groups. This is my biggest problem right now with toonsquid, I’m trying to find a workaround, if there is one, please share it! Thanks!

Im not sure if this helps but, I did this in a different way by copying and pasting them into toonsquid. https://github.com/keiwando/toonsquid/assets/135170331/9a873cbe-3d90-465b-ada0-7e2741e546fd

**Reply by VeiloPot** [reply]
If it doesnt work then im sorry.

**Comment by Cat-Butler** [comment]
Overly long rant 🙄, an over-complicated workaround, and the behaviour of some other apps worth investigating (for creating ToonSquid cutout character pieces). I hereby apologise in advance.

Unfortunately, while Procreate is a superb digital painting app, for any output beyond single images or image sequences (for e.g. a video editor) it is a terribly limited, uncooperative pain in the ... neck.

1. Even if you name all your layers in Procreate, exporting them only uses the project name, so instead of Head, Torso, etc, you get Character-1, Character-2, Character-3.

2. Procreate has the option to export all layers - hooray, right? Except when it imports into ToonSquid, you get the entire layer. If you want to use jointed/cutout rigging (as @ramzichamcham seems to be suggesting), you have to move the entire layer, as if it were a full old-fashioned animation cel, instead of just a limb. So you have to rotate an entire layer just to rotate an arm.

![arm2-sml](https://github.com/keiwando/toonsquid/assets/25494772/8adda4dc-523a-486e-8008-12885ca1d0e9)

3. That copy and paste, @VeiloPot ? Yes, as you show, great for copying/animating a single non-hierarchical piece, which is often useful - thank you for demonstrating that. Alas, not so great for setting up a hierarchical rig (a cutout-style character). 
Although the Procreate Transform tool automatically selects the layer content within Procreate, copy-and-paste from Procreate always pastes the entire layer from Procreate (see above), whether you use the Transform tool or the Layer panel select option. Even freehand selection with the Selection tool doesn't paste just the selected area. (Which is terribly frustrating, because in all these cases you can see the bit you want selected in Procreate.) So, it's back to manipulating an entire layer just to move an arm.

4. This could also potentially be the problem with importing .psd or .procreate files - oversized layered cels, rather than animation pieces, unless ToonSquid would automatically trim the layer content as part of import.

So the feature request might be extended to: trim imported images to only the non-transparent area, and if possible also keeping the relative size/position of the trimmed area.

Note: There is a (somewhat tedious) workaround for using the entire Procreate-exported/copied layers in the three points above:
- Export (or copy-paste) your layers into the ToonSquid Library (if all layers were exported from Procreate, easily done for all pieces/layers at once with a tap on **+** in the Library, Import from Files, select multiple images).
- Then, edit each full-layer piece in the Library - press/tap-and-hold on an image/layer/piece you want to edit, tap on Edit when it appears down the bottom edge of the Library window.
- Drag a manual selection around your piece, copy and paste, and you have just the piece you want to animate, automatically trimmed by ToonSquid to fit the visible area, and ready to rig. Woohoo! 😺
    - Delete the layer it was copied from, or you will still have the entire layer area.
    - Jump back to the main scene/timeline to finish editing (is there a better way to do this?)
    - Back to the Library to do the next piece.
- Back in the main timeline, a couple of clicks to add each piece (tap Library, tap the piece to add), set the pivot points, and drag them into a hierarchy - a few minutes, at most (mostly spent on carefully positioning the pivot points).
    - Unfortunately, if you have trimmed each piece (as listed above), you also need to drag it back into place after adding it to the main timeline, it will show up in the centre of the canvas. 😾

I mostly prefer working with vector apps, and my favourites all allow copying and pasting just the (trimmed) piece you want to add into ToonSquid. For example, Vectornator and Amadine make copy-and-paste into ToonSquid easy. Still a little tedious copying and pasting each piece (shapes work well, variable lines do not, convert them to outline shapes), but you get just the bits you want. However, you will need to scale them to size, move them into place, set the pivot points.

Affinity Designer allows you to copy (via Export > Copy) and also allows you to define objects as exportable "slices", allowing you to multi-export all pieces images of defined size, named and trimmed to just the pieces you want (you can also use Designer's Pixel mode for more painterly features). You can then multi-import those slice images as separate pieces, which is a step closer to ready-to-rig pieces.

**Comment by Cat-Butler** [comment]
See where I wrote:
> Drag a manual selection around your piece, copy and paste, and you have just the piece you want to animate, automatically trimmed by ToonSquid to fit the visible area, and ready to rig. Woohoo! 😺
> Delete the layer it was copied from, or you will still have the entire layer area.

That's not the only operation that will automatically trim off the large transparent parts of your imported .png image, leaving just the piece you want to animate! These are possibly even faster:
- You can Duplicate the image layer, then Merge Down, and it will auto-trim.
- You can just create a new blank layer, then Merge Down, and it will auto-trim. (Credit and Kudos to [Crittercrats, see tutorial on YouTube](https://www.youtube.com/watch?v=IZcTx_LETYU))

If you have already inserted the full-canvas symbol layer into the main timeline, you could click on the pixel layer and choose Rasterize - it will no longer be a symbol, which might not be what you want, but it will be trimmed to just the visible part for Transform purposes.

### Note: 
In my Settings, I have Animation Layer Merging set to Separate. (So I can select and animate sub-layers separately)
I have copy and paste set **not** to merge automatically.

I don't know if this will make a difference to any of the behaviour described above.

**Comment by cleesmith** [comment]
@Cat-Butler Thanks for the Crittercrats video, it was helpful.  I ended up duplicating my Procreate canvases and resizing them before adding them into ToonSquid, so they were a good size for animating otherwise copy/paste did not work well. I made a video about how I did it on [YouTube](https://www.youtube.com/live/7nxe54pHLfQ?feature=share) ... I am still learning from others.
